### PR TITLE
r.mapcalc.tiled: use parallelized r.patch if no overlap

### PR DESCRIPTION
**Pull request by petrasovaa**
This needs https://github.com/OSGeo/grass/pull/2249 but it should work for older grass versions as well.

### CODE REVIEW

**Comment by neteler** [comment]
On a sidenote: may this interesting change then also be added to the `r.texture.tiled` and `r.to.vect.tiled` addons?

**Comment by petrasovaa** [comment]
> On a sidenote: may this interesting change then also be added to the `r.texture.tiled` and `r.to.vect.tiled` addons?

Only r.texture.tiled is relevant here but the issue is r.patch as backend works only when there is no overlap, but for textures, there will be always overlaps.

**Comment by neteler** [comment]
@petrasovaa Since https://github.com/OSGeo/grass/pull/2249 has been merged, may this PR also be merged?

File: src/raster/r.mapcalc.tiled/r.mapcalc.tiled.py [hunk 1]
```diff
@@ -139,6 +190,18 @@ def main():
 
     kwargs = {"expression": expression, "quiet": True}
 
+    if not parallel_rpatch_available and patch_backend == "r.patch":
+        gs.warning(
+            _(
+                "r.patch backend is not available in this version of GRASS GIS, using RasterRow"
+            )
+        )
+        patch_backend = "RasterRow"
```
Comment (wenzeslaus) [inline src/raster/r.mapcalc.tiled/r.mapcalc.tiled.py:199]: Line not needed.

File: src/raster/r.mapcalc.tiled/r.mapcalc.tiled.py [hunk 2]
```diff
@@ -82,13 +80,33 @@
 # % required: no
 # %end
 #
+# %option
+# % key: patch_backend
+# % type: string
+# % label: Backend for patching computed tiles
+# % description: If backend is not specified, original serial implementation with RasterRow is used
+# % options: RasterRow,r.patch
+# % descriptions: RasterRow; serial patching with PyGRASS RasterRow ; r.patch; parallelized r.patch
```
Comment (wenzeslaus) [inline src/raster/r.mapcalc.tiled/r.mapcalc.tiled.py:89]: Extra space after RasterRow.

**Review by wenzeslaus — APPROVED** [review]

**Pull request merged**